hack/jenkins/e2e.sh

```diff
@@ -84,7 +82,7 @@ ${GCE_DEFAULT_SKIP_TEST_REGEX}\
 |NetworkingNew\
 |Nodes\sNetwork\
 |Nodes\sResize\
-|MaxPods\
```

Comment:
Since you're already modifying this file, I think we can also update the GCE_PARALLEL_FLAKY_TEST_REGEX. I looked into gce-parallel suite and it seems that a bunch of them haven't failed since quite a long time and we can also move them to non-flaky suite.
Can you please take a look into it and move those that didn't failed since e.g. 150 runs?